pkg/test/framework/components/echo/kube/deployment.go

```diff
@@ -235,6 +235,7 @@ spec:
         - name: EXPOSE_GRPC_ADMIN
           value: "true"
 {{- end }}
+{{- if $.ImageFullPath }}
```

Comment:
What is the relationship of image path and probe? And if there is one should it apply to liveness probe as well?

Reply:
Oops, I see the comment now. Seems like the comment says tcp vs HTTP though but this is just disabling it entirely?

Reply:
Ahhh, sorry. Forgot to push. Please take another look.

And if you have better suggestions to do this, please let me know.

Reply:
Only concern is what we want to say is "c++ doesn't support HTTP probe, so don't use it" but we end up saying "don't use HTTP probe for ImageFullPath [because we happen to use ImageFullPath for C++]".

if HTTP is not supported, what is done with the `--port` arguments? Just ignored? If so this is probably fine as is, we could add another flag but probably overkill

Reply:
Yes. We are changing readiness probe to TCP for all custom images.
We will also have a Java image later. Not sure how easy it is to start an HTTP server in Java. So this may or may not be necessary for Java as well.

Another option we have is to change readiness probe to prefer gRPC whenever there is a gRPC server.

And right, the `--port` arguments are ignored by the C++ image right now.